app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,12 @@
+class OrdersController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+
+  def index
+    @index_presenter = Orders::IndexPresenter.new(current_user.orders, params)
+  end
+
+  def show
+    @order = current_user.orders.find_by!(id: params[:id])
```

Comment:
the same as for books_controller
Also if you want to raise 404 you can just use `find` method
https://medium.com/@alroth/find-vs-find-by-1518df5c657d